### ISSUE TITLE
Add excerpt to the index

### ISSUE THREAD

**Issue by piscis** [algolia/algoliasearch-wordpress#37]
Please consider to add the excerpt back to the index by default.

It would be nice if a excerpt could be added to the index at the moment we have to do this manually by commenting in this line of code.

https://github.com/algolia/algoliasearch-wordpress/blob/c157fb7a61141b0a995c8208e74df8a60f94e1b9/core/WordpressFetcher.php#L217

Could you elaborate why this was commented out?

**Comment by maxiloc** [comment]
Yes, We implement a way better better strategy to handle content by cutting the html into pieces. If you don't want to use this strategy, using attributesToSnippet with content is also better.

**Comment by maxiloc** [comment]
I am re-opening this one. We need to find a better way to handle use cases like #52

**Comment by maxiloc** [comment]
@crebacz @piscis Can you tell me a bit more details on how you use excerpt. If I get the $post->post_excerpt value, would that be good for you ?

**Comment by crebacz** [comment]
@maxiloc,  I was going to reach out to let you know how I was able to address the issue from my end.  

To answer your first question, whenever we add an article/post, we add a custom excerpt from within the post-edit screen that is saved alongside the post content.  It is a 1-2 sentence explanation of what type of information and the format that the content is presented in (video, diagram, written content, audio etc). If this custom excerpt is not added for some reason, we rely on the automated excerpt generation from within Wordpress for the first 24 words of an article.  

To answer your second question, the $post->post_excerpt value will definitely work for our needs.  In fact, what I did to get the excerpt working is: 
In WordpressFetcher.php
1. I removed the unsetting of the post->post_excerpt at line 217.
2. I uncommented line 218, and modified the parameters of the my_excerpt functions to include the post->post_excerpt rather than the get_the_excerpt function: 

```
$obj->excerpt           = my_excerpt($data->post_content, $data->post_excerpt());

```

In the templates.php, I modified the #instant_content_template to pull the data from {{excerpt}} rather than {{{_highlightResult.content.value}}}.  

There are two downsides to my approach: 
1. This always saves an excerpt value in Algolia, which (to your point in issue #37) is in most cases redundant, and hinders your plugin's excerpt generation functionality. Just spitballing, but what if I removed the call to the my_excerpt function.  It would save an empty string for the excerpt, allowing for a conditional within the template (ie, if excerpt === '', pass the _highlightResult.content.value to the template instead).  
2. I was unable to figure out how to add excerpt to the list of attributes that would be included in the _highlightResult, and so as it stands, it does not offer the highlighting that your solution has by default.  While this isn't a big deal for my particular use case, I could see it as a problem in other situations.  

Hope this helps.

**Comment by maxiloc** [comment]
@crebacz Thanks for your quick feedback, much appreciated. 

1. will be fix in release 1.0 as every attribute will need to be specified to be send and excerpt will not be in the default configuration.

2. By default the highlighting should be enable for all attributes. The plugin do not set any so all have it. The only case where it is possible not to have highlighting is a value was set manually on the dashboard.
So it will not be an issue either.

It is fixed on develop branch. So I am closing this.
